### PR TITLE
Adjust ObjectBucket create doc page

### PR DESCRIPTION
**Pull request by zugao**
I had to adjust the documentation for ObjectBucket as it looked wrong. The remark was unnecessary and there was also a factual error.

### CODE REVIEW

File: docs/modules/ROOT/pages/object-storage/create.adoc [hunk 1]
```diff
@@ -20,11 +20,11 @@ spec:
     name: objectbucket-creds # <4>
 ----
 <1> The namespace where the object will be created.
-<2> Bucket name: *don't just copy/paste this code, remember to choose your own name here!*
+<2> The bucket name for this ObjectBucket.
 <3> The bucket's region depends on the selected Provider, see xref:references/cloud-zones.adoc#_regions[here] for more details
 <4> Secret where the connection details are provisioned. This secret shouldn't exist at this time.
 +
-IMPORTANT: Object Storage Bucket names must be unique across providers. Make sure to choose a name with prefixes and other guards, to prevent any issue during the creation of the bucket. No error message is returned if the bucket cannot be created.
```
Comment (wejdross) [inline docs/modules/ROOT/pages/object-storage/create.adoc:27]: `No error message is returned if the bucket cannot be created.`

Why did You delete that? Looks like important docs for me

**Review by wejdross — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/object-storage/create.adoc (hunk 1 above)
Comment (zugao) [reply docs/modules/ROOT/pages/object-storage/create.adoc:27]: I'm almost sure we return an error if the bucket is not created directly in the claim.

Comment (akosma) [reply docs/modules/ROOT/pages/object-storage/create.adoc:27]: 2 things:

1. About "almost sure": we need to be sure whether this happens or not.
2. Technically, when you `oc apply` the YAML for a bucket, there's no error printed to the user on the terminal, or on the web console, so, in terms of developer experience, no, there is no error. And developers are the primary audience of this documentation. Yes, I understand the "they should know where to find it" but the whole point of this documentation set is to remove friction.

I do not approve removing this text from the page because doing so increases the friction of using AppCat. Having said that, if the error is indeed stored in the claim, then we should rephrase the text including the required command or procedure to retrieve the error, asking users to proactively check for this situation in case their bucket is not provisioned correctly after a while.

Comment (wejdross) [reply docs/modules/ROOT/pages/object-storage/create.adoc:27]: can't we just introduce webhook to increase UX? just reject claim if name already exists? I know that I'm boring talking constantly about webhooks, but we must increase UX constantly, otherwise we'll produce nice product that no one wants to use

Comment (zugao) [reply docs/modules/ROOT/pages/object-storage/create.adoc:27]: > 2 things:
> 
> 1. About "almost sure": we need to be sure whether this happens or not.
> 2. Technically, when you `oc apply` the YAML for a bucket, there's no error printed to the user on the terminal, or on the web console, so, in terms of developer experience, no, there is no error. And developers are the primary audience of this documentation. Yes, I understand the "they should know where to find it" but the whole point of this documentation set is to remove friction.
> 
> I do not approve removing this text from the page because doing so increases the friction of using AppCat. Having said that, if the error is indeed stored in the claim, then we should rephrase the text including the required command or procedure to retrieve the error, asking users to proactively check for this situation in case their bucket is not provisioned correctly after a while.

1. You are right I will do some tests to be 100% sure.
2. There cannot be such errors displayed right away because deploying a service requires time even for object buckets. It make take sometimes up to a minute to make the bucket Ready due to many "players" involved. The customer has to check the claim afterwards with `describe` or `get` operation. 

> can't we just introduce webhook to increase UX?
The webhook is made to only validate the input, it cannot validate the process from the backend because the webhook is in front.

Comment (akosma) [reply docs/modules/ROOT/pages/object-storage/create.adoc:27]: I understand we can't have a synchronous operation here the insane number of intermediates, but then we have to document which errors could appear, and how to find the status of the request, wherever it is.

Comment (zugao) [reply docs/modules/ROOT/pages/object-storage/create.adoc:27]: > I understand we can't have a synchronous operation here given the insane number of intermediates, but then we have to document which errors could appear, and how to find the status of the request, wherever it is.

Absolutely agree.

Comment (zugao) [reply docs/modules/ROOT/pages/object-storage/create.adoc:27]: I just finished a test:

```
Bucket Conditions:
    Last Transition Time:  2023-10-26T11:05:13Z
    Message:               observe failed: bucket exists already, try changing bucket name: exoscale-bucket-lab-test-gsaratura
    Reason:                ReconcileError
    Status:                False
    Type:                  Synced
```
Doing `describe` or `get` operation will reveal what is the problem, in this case the bucket in the backend already exists. So the backend errors are being propagated to the claim. One thing we can improve is to copy these errors to `events` as well so that we don't have just:

```
Normal   BindCompositeResource       2m5s (x9 over 2m15s)  offered/compositeresourcedefinition.apiextensions.crossplane.io  Composite resource is not yet ready
```

Comment (zugao) [reply docs/modules/ROOT/pages/object-storage/create.adoc:27]: I added a comment here

Comment (Kidswiss) [reply docs/modules/ROOT/pages/object-storage/create.adoc:27]: > One thing we can improve is to copy these errors to events as well so that we don't have just:

Afaik we have a ticket since basically the beginning to implement something like that. Maybe it's finally time that this should be tackled? Not only specific for the objectbuckets though, more generally for all the services.

File: docs/modules/ROOT/pages/object-storage/create.adoc [hunk 2]
```diff
@@ -20,11 +20,11 @@ spec:
...
+IMPORTANT: Object Storage Bucket names must be unique across providers. Make sure to choose a name with prefixes and other guards, to prevent any issue during the creation of the bucket. Check the status field of the custom resource for such errors.
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/object-storage/create.adoc:27]:
```suggestion
IMPORTANT: Object Storage Bucket names must be unique across providers. Make sure to choose a name with prefixes and other guards, to prevent any issue during the creation of the bucket. Check the status field of the `ObjectBucket` for such errors.
```

**Review by Kidswiss — APPROVED** [review]
